pkg/validator/validator_test.go

```diff
@@ -216,6 +221,57 @@ func TestValidateOperationalOptions(t *testing.T) {
 	}
 }
 
+func TestValidateFile(t *testing.T) {
+	tests := map[string]struct {
+		path        string
+		wantError   bool
+		notReadable bool
+		expected    string
+	}{
+		"existing file": {
+			path: "../../testdata/inputs/empty.txt",
+		},
+		"unexisting file": {
+			path:      "../../testdata/inputs/this_doesnt_exist.txt",
+			wantError: true,
+			expected:  "file '../../testdata/inputs/this_doesnt_exist.txt' does not exist",
+		},
+		"directory": {
+			path:      "../../testdata/inputs",
+			wantError: true,
+			expected:  "file '../../testdata/inputs' is a directory",
+		},
+		"not readable": {
+			path:        "../../testdata/inputs/noperms.txt",
+			wantError:   true,
+			expected:    "file '../../testdata/inputs/noperms.txt' is not readable",
+			notReadable: true,
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			if test.notReadable {
+				err := os.Chmod(test.path, 0222)
+				if err != nil {
+					t.Fatalf("Failed to change permissions of file %s: %s", test.path, err.Error())
+				}
+			}
```

Comment:
This causes some troubles with git, for example checking out another branch fails:
```console
% git checkout master
error: Your local changes to the following files would be overwritten by checkout:
	testdata/inputs/noperms.txt
Please commit your changes or stash them before you switch branches.
Aborting
```
What do you think about setting these permissions directly on the file, given Git should preserve them?

Reply:
That was my initial solution but, for some reason, the Github Actions CI wasn't preserving them (I though it could be something like root perms when running the tests but changing them in the code fixed it). I can try to find another way, since this approach doesn't work well either

Reply:
It seems like git actually doesn't preserve them upon checkout and it won't keep them for you: https://stackoverflow.com/questions/14557106/git-changes-my-file-permissions-upon-checkout

It seems we have to do it manually or remove this test. Any thoughts?

Reply:
Fixed by creating a temporary file instead